framework/src/services.py

```diff
@@ -2366,12 +2380,12 @@ def create_sub_workflow(self, sub_name, config_file, override= None, input_dir=
         ports = sub_conf_new['PORTS']['NAMES'].split()
         comps = [sub_conf_new['PORTS'][p]['IMPLEMENTATION'] for p in ports]
         for c in comps:
-            if not c:
```

Comment:
Why is this deleted? This check is needed since sometimes a listed port may not have an implementation, which is fine as long as the driver doesn't use it

Reply:
Thats a good question. Probably shouldn't be.